src/main/java/net/openhft/chronicle/threads/Threads.java

```diff
@@ -167,8 +169,8 @@ public static void interrupt(ExecutorService service) {
     static void forEachThread(ExecutorService service, Consumer<Thread> consumer) {
         try {
             final Set workers;
-            if (Jvm.isJava9Plus() && !(service instanceof ThreadPoolExecutor)) {
-                workers = Jvm.getValue(service, "e/workers");
+            if (!(service instanceof ThreadPoolExecutor)) {
+                workers = Jvm.getValue(resolveDelegatedExecutorServices(service), "workers");
```

Comment:
I'm not sure why this was only for Java 9 plus. DelegatedExecutorService exists in 8 too.

Reply:
Also now we resolve arbitrary depths of delegation. I wonder if that should be limited in case of a cycle.

Reply:
definitely good to have less `if (Jvm.isJava9Plus())` s